examples/json/README.md

```diff
@@ -0,0 +1,164 @@
+# JSON example
+
+This example demonstrates writing a function that accepts a JSON request and
+sends a JSON response to the client.
+
+The basic shape of the function handler looks like this:
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
+}
+```
+
+The client will send a JSON document using the HTTP POST method. Here's an
+example request:
+
+```json
+{
+  "name": "World"
+}
+
+```
+
+The function will send a JSON document as the response. Here's an example
+response after receiving the above request:
+
+```json
+{
+  "salutation": "Hello",
+  "name": "World"
+}
+```
+
+The function handler actually receives the JSON request parsed as a map by the
+Functions Framework. To access the expected `name` field of the request
+document (and provide a default value in case the `name` field is empty), the
+code looks like this:
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
+  final name = request['name'] as String ?? 'World';
+}
+```
+
+Finally, the function creates an object returns it. The Functions Framework will
+take this and attempt "do the right" thing. In this case, the function is typed
+to return a `FutureOr<GreetingResponse>`, which has a `toJson()` method, so the
+framework will invoke this, then set the response body to the stringified result
+and set the response header (`content-type`) to `application/json`).
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
+  final name = request['name'] as String ?? 'World';
+  final json = GreetingResponse(salutation: 'Hello', name: name);
+  return json;
+}
+```
+
+The full code is shown below:
+
+_lib/functions.dart_
```

Comment:
the markdown linter doesn't like this 🤷

Reply:
The linter is wrong! I don't like how it converted the filename to a Dart comment for `example/hello` 😖

Reply:
🤷

Reply:
Grateful to the linter for finding my link mistakes at least! Hang on a moment.